mempool/mempool.go

```diff
@@ -162,7 +164,7 @@ func (mem *Mempool) TxsFrontWait() *clist.CElement {
 // It blocks if we're waiting on Update() or Reap().
 // cb: A callback from the CheckTx command.
 //     It gets called from another goroutine.
-// CONTRACT: Either cb will get called, or err returned.
+// NOTE: Either cb will get called, or err returned.
```

Comment:
why this changed? things which can't be captured in the function signature should be made explicit in a `CONTRACT` comment. We should add that to the coding repo @zramsay

Reply:
Oh, I thought we put those under `NOTE:`. The nice thing with `NOTE`, `TODO` or `XXX` is that emacs highlights those.

Reply:
yeh the highlighting is nice. we can discuss this. but NOTE feels too informal

Reply:
Actually, let's stick with `CONTRACT` for now. I'll edit my emacs to also highlight that.